### PR TITLE
Move app to packages/studio-base/src

### PR DESCRIPTION
**Pull request by amacneil**
There are two commits in this PR. The first one is mechanical (rename files) and can be easily recreated. The second commit contains actual code modifications.

Please keep this PR easy to rebase in the future, by amending any code changes to the second commit. Don't add additional commits.

How to rebase this PR (trash and recreate the first commit, then cherry pick the second commit):

```sh
# start from a fresh master
git status  # start with a clean local tree
git fetch
git reset --hard origin/main

# create the rename commit
mkdir -p packages/studio-base && \
git mv app packages/studio-base/src && \
git mv packages/studio-base/src/.eslintrc.yaml packages/studio-base/ && \
git mv packages/studio-base/src/*.json packages/studio-base/ && \
git mv packages/studio-base/src/webpack* packages/studio-base/ && \
git mv packages/studio-base/src/Webpack* packages/studio-base/ && \
git commit -m 'Rename files'

# cherry pick the actual code changes
git cherry-pick <sha>  # use the commit sha from the second commit in this PR
# fix any merge conflicts
```

### CODE REVIEW

**Comment by defunctzombie** [comment]
.storybook should not be in src

**Comment by defunctzombie** [comment]
@amacneil still planning to land this?

File: packages/studio-base/src/test/transformers/typescriptTransformerWithRawImports.js [hunk 1]
```diff
@@ -5,25 +5,24 @@
...
 const importRegEx = /^import (.*) from "(.*)\?raw";$/gm;
-const importReplacer = (_, p1, p2) => {
-  const resolved = require.resolve(p2);
-  const rawFile = fs.readFileSync(resolved, { encoding: "utf-8" });
-  return `const ${p1} = ${JSON.stringify(rawFile.toString())};`;
-};
-
-function rewriteSource(source) {
-  return source.replace(importRegEx, importReplacer);
+function rewriteSource(source, sourcePath) {
```
Comment (defunctzombie) [inline packages/studio-base/src/test/transformers/typescriptTransformerWithRawImports.js:13]: I made a separate PR for these changes so we can minimize the second commit diff.
https://github.com/foxglove/studio/pull/1136

File: packages/studio-base/webpack.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ export function makeConfig(
     resolve: {
       extensions: [".js", ".ts", ".jsx", ".tsx"],
       alias: {
+        "@foxglove/studio-base": path.resolve(__dirname, "src"),
```
Comment (jtbandes) [inline packages/studio-base/webpack.ts:46]: Why are these changes needed now?

Comment (amacneil) [reply packages/studio-base/webpack.ts:46]: because we want to avoid the extra `src` directory in imports. previously this was not needed because the paths exactly matched the folder structure
